massa-storage/src/endorsement_indexes.rs

```diff
@@ -21,15 +21,16 @@ impl EndorsementIndexes {
     /// Arguments:
     /// - endorsement: the endorsement to insert
     pub(crate) fn insert(&mut self, endorsement: SecureShareEndorsement) {
-        if let Ok(e) = self
-            .endorsements
-            .try_insert(endorsement.id, Box::new(endorsement))
-        {
+        if !self.endorsements.contains_key(&endorsement.id) {
```

Comment:
this does hashing and search twice, maybe something like that:

```rust
if let(Vacant(mut vac)) = self.endorsements.entry(endorsement.id) {
           let endorsement = vac.insert(Box::new(enforsement));
           self.index_by_creator
                .entry(endorsement.content_creator_address)
                .or_default()
                .insert(endorsement.id);
        massa_metrics::set_endorsements_counter(self.endorsements.len());
}
```

Reply:
Done